worker/src/services/spotify.ts

```diff
@@ -0,0 +1,69 @@
+import SpotifyWebApi from 'spotify-web-api-js';
+import { Pipeline, PipelineType, ReactionType, ServiceType } from "../models/pipeline";
+import { BaseService, reaction, service } from "../models/base-service";
+
+@service(ServiceType.Spotify)
+export class Spotify extends BaseService {
+
+	private _spotify;
+
+	constructor(_: Pipeline) {
+		super();
+		// TODO load credentials
+		this._spotify = new SpotifyWebApi();
+	}
+
+	private async searchTrack(artistName: string, trackName: string)  {
+		return this._spotify.searchTracks(`artist=${artistName}&track=${trackName}`).then(
```

Comment:
You should not return a Promise with a then in an async's body. Simply use await.